bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2495,23 +2495,22 @@ void ReplicaImp::sendCheckpointIfNeeded() {
   sendToAllOtherReplicas(checkpointMessage, true);
 }
 
-void ReplicaImp::onTransferringCompleteImp(SeqNum newStateCheckpoint) {
+void ReplicaImp::onTransferringCompleteImp(uint64_t newStateCheckpoint) {
   TimeRecorder scoped_timer(*histograms_.onTransferringCompleteImp);
-  ConcordAssertEQ(newStateCheckpoint % checkpointWindowSize, 0);
 
   LOG_INFO(GL, KVLOG(newStateCheckpoint));
 
   if (ps_) {
     ps_->beginWriteTran();
   }
-
-  if (newStateCheckpoint <= lastExecutedSeqNum) {
+  SeqNum newCheckpointSeqNum = newStateCheckpoint * checkpointWindowSize;
+  if (newCheckpointSeqNum <= lastExecutedSeqNum) {
     LOG_DEBUG(GL,
               "Executing onTransferringCompleteImp(newStateCheckpoint) where newStateCheckpoint <= lastExecutedSeqNum");
```

Comment:
nit - you could maybe add a KVLOG here for newCheckpointSeqNum and lastExecutedSeqNum